src/libclient/export/videoexporter.h

```diff
@@ -144,4 +119,11 @@ class VideoExporter : public QObject
 	QSize _targetsize;
 };
 
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=69210
+namespace diagnostic_marker_private {
```

Comment:
Per chat discussion, for posterity: This is because MSVC does not know `#pragma GCC` and will complain, clang does know `#pragma GCC` but not `-Wsuggest-final-types`/`methods`, you can’t use parameterised macros for pragmas, and so macroing around all of this would require having a new file with the macros in it, one for each diagnostic, plus sending the feature detection flag from CMake into the compiler to enable it, and that just seems like a lot of extra machinery spread out everywhere versus using this weird but fully encapsulated way to suppress the false positive diagnostics. It’s not great but it is what it is.